subprojects/wrapper-shared/src/test/groovy/org/gradle/wrapper/InstallTest.groovy

```diff
@@ -16,56 +16,80 @@
 
 package org.gradle.wrapper
 
+
 import org.gradle.test.fixtures.file.CleanupTestDirectory
 import org.gradle.test.fixtures.file.TestFile
 import org.gradle.test.fixtures.file.TestNameTestDirectoryProvider
+import org.junit.ClassRule
 import org.junit.Rule
+import spock.lang.Shared
 import spock.lang.Specification
 
 import java.util.zip.ZipEntry
 import java.util.zip.ZipOutputStream
 
 @CleanupTestDirectory
 class InstallTest extends Specification {
+    public static final String FETCHED_HASH = "fetchedHash"
+    WrapperConfiguration configuration = new WrapperConfiguration()
+
+    private final String zipFileName = 'gradle-0.9.zip';
+
     File testDir
-    Install install
-    boolean downloadCalled
-    File zip
     TestFile distributionDir
-    File zipStore
     TestFile gradleHomeDir
+    File zipStore
     TestFile zipDestination
-    WrapperConfiguration configuration = new WrapperConfiguration()
+
     IDownload download = Mock()
     PathAssembler pathAssembler = Mock()
     PathAssembler.LocalDistribution localDistribution = Mock()
+    Install install
     @Rule
     public TestNameTestDirectoryProvider temporaryFolder = new TestNameTestDirectoryProvider(getClass())
 
+    @Shared @ClassRule TestNameTestDirectoryProvider commonTemporaryFolder = TestNameTestDirectoryProvider.newInstance(getClass())
```

Comment:
Maybe use "shared" instead of "common"? I didn't understand what it was supposed to mean, until I noticed the `@ClassRule` annotation.